dds-test/src/unitary/conversion.cpp

```diff
@@ -0,0 +1,471 @@
+    REQUIRE(soss_data["my_float32"].value<float>() == 55.555e3f);
+    REQUIRE(soss_data["my_float64"].value<double>() == 5.8598e40);
+    REQUIRE(soss_data["my_float128"].value<long double>() == 3.54e2400l);
+    REQUIRE(soss_data["my_char"].value<char>() == 'P');
+    REQUIRE(soss_data["my_wchar"].value<wchar_t>() == L'G');
+    REQUIRE(soss_data["my_string"].value<std::string>() == "Testing a string.");
+    REQUIRE(soss_data["my_wstring"].value<std::wstring>() == L"Testing a wstring: \u20B1");
+}
+
+static void fill_nested_sequence(
+        soss::WritableDynamicDataRef soss_data)
+{
+    const soss::SequenceType& inner_type = static_cast<const soss::SequenceType&>(soss_data.type());
+    for (size_t i = 0; i < 3; ++i)
+    {
+        soss::DynamicData inner_seq(inner_type.content_type());
+        for (size_t j = 0; j < 2; ++j)
+        {
+            int32_t temp = j + (i * 2);
+            inner_seq.push(temp);
+        }
+        soss_data.push(inner_seq);
+    }
+}
+
+static void check_nested_sequence(
+        dds::DynamicData* dds_data)
+{
+    for (uint32_t i = 0; i < 3; ++i)
+    {
+        dds::DynamicData* inner_seq= dds_data->loan_value(i);
+        for (uint32_t j = 0; j < 2; ++j)
+        {
+            int32_t temp = inner_seq->get_int32_value(j);
+            REQUIRE(temp == (j + (i * 2)));
+        }
+        dds_data->return_loaned_value(inner_seq);
+    }
+}
+
+static void check_nested_sequence(
+        soss::ReadableDynamicDataRef soss_data)
+{
+    for (uint32_t i = 0; i < 3; ++i)
+    {
+        for (uint32_t j = 0; j < 2; ++j)
+        {
+            int32_t temp = soss_data[i][j];
+            REQUIRE(temp == (j + (i * 2)));
+        }
+    }
+}
+
+static void fill_nested_array(
+        soss::WritableDynamicDataRef soss_data)
+{
+    for (size_t i = 0; i < 4; ++i)
+    {
+        for (size_t j = 0; j < 5; ++j)
+        {
+            int32_t temp = j + (i * 5);
+            soss_data[i][j] = std::to_string(temp);
+        }
+    }
+}
+
+static void check_nested_array(
+        dds::DynamicData* dds_data)
+{
+    for (uint32_t i = 0; i < 4; ++i)
+    {
+        for (uint32_t j = 0; j < 5; ++j)
+        {
+            dds::MemberId idx = dds_data->get_array_index({i, j});
+            std::string temp = dds_data->get_string_value(idx);
+            REQUIRE(temp == std::to_string(j + (i * 5)));
+        }
+    }
+}
+
+static void check_nested_array(
+        soss::ReadableDynamicDataRef soss_data)
+{
+    for (uint32_t i = 0; i < 4; ++i)
+    {
+        for (uint32_t j = 0; j < 5; ++j)
+        {
+            std::string temp = soss_data[i][j];
+            REQUIRE(temp == std::to_string(j + (i * 5)));
+        }
+    }
+}
+
+static void fill_mixed_struct(
+        soss::DynamicData& soss_data)
+{
+    using namespace soss;
+    const StructType& mixed_type = static_cast<const StructType&>(soss_data.type());
+    //sequence<BasicStruct, 3> my_basic_seq;
```

Comment:
Is this code commented or a real comment?

Reply:
It's a real comment which helps to understand the code following it.